contracts/examples/bonding-curve/src/lib.rs

```diff
@@ -0,0 +1,325 @@
+			.mint(&token, &amount)
+			.async_call()
+			.with_callback(self.callbacks().mint_callback(&caller, &amount)))
+	}
+
+	#[callback]
+	fn mint_callback(
+		&self,
+		caller: &Address,
+		amount: &Self::BigUint,
+		#[call_result] result: AsyncCallResult<()>,
+	) {
+		match result {
+			AsyncCallResult::Ok(()) => {
+				self.mint_success_event(caller);
+				self.supply().update(|supply| *supply += amount);
+				self.balance().update(|balance| *balance += amount);
+			},
+			AsyncCallResult::Err(message) => {
+				self.mint_failure_event(caller, message.err_msg.as_slice());
+			},
+		}
+	}
+
+	// endpoints
+
+	fn check_sell_requirements(
+		&self,
+		token: &TokenIdentifier,
+		amount: &Self::BigUint,
+	) -> SCResult<()> {
+		let issued_token = self.issued_token().get().identifier;
+		if &issued_token != token {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Only ",
+				token.as_esdt_identifier(),
+				b" tokens accepted",
+			])));
+		}
+		if self.balance().get() < amount.clone() {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Contract does not have enough ",
+				token.as_esdt_identifier(),
+				b". Please try again once more is minted.",
+			])));
+		}
+		require!(
+			amount > &Self::BigUint::zero(),
+			"Must pay more than 0 tokens!"
+		);
+		Ok(())
+	}
+
+	fn check_buy_requirements(
+		&self,
+		token: &TokenIdentifier,
+		amount: &Self::BigUint,
+	) -> SCResult<()> {
+		let exchanging_token = &self.exchanging_token().get().identifier;
+		if exchanging_token != token {
+			return Err(SCError::from(BoxedBytes::from_concat(&[
+				b"Only ",
+				exchanging_token.as_esdt_identifier(),
+				b" tokens accepted",
+			])));
+		}
+		require!(
+			amount > &Self::BigUint::zero(),
+			"Must pay more than 0 tokens!"
+		);
+		Ok(())
+	}
+	fn get_curve_arguments(self) -> CurveArguments<Self::BigUint> {
+		CurveArguments {
+			supply_type: self.supply_type().get(),
+			max_supply: self.max_supply().get(),
+			current_supply: self.supply().get(),
+			balance: self.balance().get(),
+		}
+	}
+
+	fn get_function(&self) -> LinearFunction<Self::BigUint> {
+		LinearFunction {
+			a: Self::BigUint::from(3u64),
+			b: Self::BigUint::from(5u64),
+		}
+	}
+
+	fn calculate_buy_price(&self, amount: Self::BigUint) -> SCResult<Self::BigUint> {
+		self.get_function().buy(amount, self.get_curve_arguments())
+	}
+
+	fn calculate_sale_price(&self, amount: Self::BigUint) -> SCResult<Self::BigUint> {
+		self.get_function().sell(amount, self.get_curve_arguments())
+	}
+
+	#[payable("*")]
+	#[endpoint(buyToken)]
+	fn buy_token(
+		&self,
+		#[payment] buy_amount: Self::BigUint,
+		#[payment_token] token_identifier: TokenIdentifier,
+	) -> SCResult<()> {
+		self.check_buy_requirements(&token_identifier, &buy_amount)?;
+
+		let calculated_price = &self.calculate_buy_price(buy_amount.clone()).unwrap();
+
+		self.balance().update(|balance| *balance += buy_amount);
+
+		let caller = self.blockchain().get_caller();
+
+		let _ = self
```

Comment:
self.send().direct() returns nothing, there is no need to have the "let _" in front